schema_editor/app/scripts/notifications/notifications-service.js

```diff
@@ -0,0 +1,63 @@
+(function () {
+    'use strict';
+
+    /* ngInject */
+    function Notifications($rootScope, $timeout) {
+
+        var timeoutId = null;
+        var active = null;
+
+        var module = {
+            hide: hide,
+            show: show,
+            activeAlert: activeAlert
+        };
+
+        return module;
+
+        /**
+         * Provide the currently active alert.
+         * Allows directives to display the active alert even if they miss the relevant event
+         * @return the currently active alert
+         */
+        function activeAlert() {
+            return active;
+        }
+
+        /**
+         * Hide a global notification
+         * @return Broadcasts ase.notifications.hide on $rootScope
+         */
+        function hide() {
+            if (timeoutId) {
+                $timeout.cancel(timeoutId);
+                timeoutId = null;
+            }
+            active = null;
+            $rootScope.$broadcast('ase.notifications.hide');
+        }
+
+        /**
+         * Show a global notification, overridding the defaults defined in the function
+         * @param  {object} options Override the defaults with this config
+         * @return Broadcasts ase.notifications.show on $rootScope
+         */
+        function show(options) {
+            var defaults = {
+                timeout: 0,
+                top: '8rem',
```

Comment:
It doesn't look like this is actually being used, so it can probably be removed. If we do need something like this, I think it would be better handled in css anyway.